application/classes/Ushahidi/Repository/Dataprovider.php

```diff
@@ -28,6 +28,25 @@ class Ushahidi_Repository_Dataprovider implements
 	// ReadRepository
 	public function getEntity(Array $data = null)
 	{
+		// translating labels and description for data-providers
+		if($data !== null && $data['options']) {
+			foreach ($data['options'] as $option_key => $option) {
+				foreach($option as $value_key => $value) {
+					if($value_key === 'label' && $value !== '') {
+						$data['options'][$option_key]['label'] = __($value);
+					}
+					if($value_key === 'description') {
+						$description = $data['options'][$option_key]['description'];
+						if(gettype($description) === 'string') {
+							$data['options'][$option_key]['description'] = __($value);
```

Comment:
These long variables doesn't look nice but I couldn't find another way of doing it. Assigning $description with the new value doesn't assign it to the $data-variable.